modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,76 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.MonadThrow
+import cats.syntax.all._
+import org.tessellation.schema.SnapshotOrdinal
+import derevo.cats.eqv
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+import scala.util.control.NoStackTrace
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: List[HistoricalSnapshotOrdinalEntry])
+
+case object InvalidRange extends NoStackTrace
+
+object HistoricalSnapshotOrdinalEntries {
+
+  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
+    CellDecoder.stringDecoder.emap { s =>
+      mkSnapshotOrdinalTimeline(s).attempt
+        .map(
+          _.leftMap(_ => new DecoderError("Unable to parse the ordinal timeline"))
+            .map(_.some)
+        )
+        .get
```

Comment:
Why you need to explicitly run `.get` and unwrap sth? Looks suspicious

Reply:
Reworked to

```scala
  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
    CellDecoder.stringDecoder.emap { s =>
      mkSnapshotOrdinalTimeline(s)
        .leftMap(_ => new DecoderError("Unable to parse the ordinal timeline"))
        .map(_.some)
    }
```